src/phase.cs

```diff
@@ -0,0 +1,1299 @@
+
+		[Export ("activateWithTimeToTargetOverride:")]
+		void ActivateWithTimeToTargetOverride (double timeToTargetOverride);
+
+		[Export ("deactivate")]
+		void Deactivate ();
+
+		[Export ("deactivateWithTimeToResetOverride:")]
+		void DeactivateWithTimeToResetOverride (double timeToResetOverride);
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASEMedium")]
+	[DisableDefaultCtor]
+	interface PhaseMedium
+	{
+		[Export ("initWithEngine:preset:")]
+		IntPtr Constructor (PhaseEngine engine, PhaseMediumPreset preset);
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASEObject")]
+	[DisableDefaultCtor]
+	interface PhaseObject : NSCopying
+	{
+		[Export ("initWithEngine:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (PhaseEngine engine);
+
+		[Export ("addChild:error:")]
+		bool AddChild (PhaseObject child, [NullAllowed] out NSError error);
+
+		[Export ("removeChild:")]
+		void RemoveChild (PhaseObject child);
+
+		[Export ("removeChildren")]
+		void RemoveChildren ();
+
+		[NullAllowed, Export ("parent", ArgumentSemantic.Weak)]
+		PhaseObject Parent { get; }
+
+		[Export ("children", ArgumentSemantic.Copy)]
+		PhaseObject[] Children { get; }
+
+/*
+		[Static]
+		[Export ("right")]
+		[unsupported ExtVector: float __attribute__((ext_vector_type(3)))] Right { get; }
+
+		[Static]
+		[Export ("up")]
+		[unsupported ExtVector: float __attribute__((ext_vector_type(3)))] Up { get; }
+
+		[Static]
+		[Export ("forward")]
+		[unsupported ExtVector: float __attribute__((ext_vector_type(3)))] Forward { get; }
+
+		[Export ("transform", ArgumentSemantic.Assign)]
+		simd_float4x4 Transform { get; set; }
+
+		[Export ("worldTransform", ArgumentSemantic.Assign)]
+		simd_float4x4 WorldTransform { get; set; }
+*/
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASESoundEvent")]
+	[DisableDefaultCtor]
+	interface PhaseSoundEvent
+	{
+		[Export ("initWithEngine:assetIdentifier:mixerParameters:error:")]
+		IntPtr Constructor (PhaseEngine engine, string assetIdentifier, PhaseMixerParameters mixerParameters, [NullAllowed] out NSError error);
+
+		[Export ("initWithEngine:assetIdentifier:error:")]
+		IntPtr Constructor (PhaseEngine engine, string assetIdentifier, [NullAllowed] out NSError error);
+
+		[Async]
+		[Export ("prepareWithCompletionBlock:")]
+		void Prepare (Action<PhaseSoundEventPrepareHandlerReason> completionBlock);
+
+		[Export ("prepareAndReturnError:")]
+		bool Prepare ([NullAllowed] out NSError error);
+
+		[Async]
+		[Export ("startWithCompletionBlock:")]
+		bool Start (Action<PhaseSoundEventStartHandlerReason> completionBlock);
+
+		[Export ("startAndReturnError:")]
+		bool Start ([NullAllowed] out NSError error);
+
+		[Export ("seekToTime:completionBlock:")]
+		bool SeekToTime (double time, [NullAllowed] Action<PhaseSoundEventSeekHandlerReason> completionBlock);
```

Comment:
```suggestion
		bool Seek (double time, [NullAllowed] Action<PhaseSoundEventSeekHandlerReason> completionBlock);
```
Also would this one be a candidate for async?